src/main/java/mekanism/common/item/interfaces/IJetpackItem.java

```diff
@@ -113,33 +117,38 @@ private static ItemStack getJetpack(LivingEntity entity, Predicate<ItemStack> ma
     /**
      * @return If fall distance should get reset or not
      */
-    static boolean handleJetpackMotion(Player player, JetpackMode mode, BooleanSupplier ascendingSupplier) {
+    static boolean handleJetpackMotion(Player player, JetpackMode mode, double thrust, BooleanSupplier ascendingSupplier) {
         Vec3 motion = player.getDeltaMovement();
-        if (mode == JetpackMode.NORMAL) {
-            if (player.isFallFlying()) {
-                Vec3 forward = player.getLookAngle();
-                Vec3 delta = forward.multiply(forward.scale(0.15))
-                      .add(forward.scale(1.5).subtract(motion).scale(0.5));
-                player.setDeltaMovement(motion.add(delta));
-                return false;
-            } else {
-                player.setDeltaMovement(motion.x(), Math.min(motion.y() + 0.15D, 0.5D), motion.z());
-            }
+        if ((mode == JetpackMode.NORMAL || mode == JetpackMode.VECTOR) && player.isFallFlying()) {
+            Vec3 forward = player.getLookAngle();
+            Vec3 drag = forward.scale(1.5).subtract(motion).scale(0.5);
+            Vec3 delta = forward.scale(thrust).add(drag);
```

Comment:
You mentioned on discord a bug your discovered with the elytra interaction. Was the issue basically the `forward.multiply(forward.scale(thrust))` vs `forward.scale(thrust)`? And if that is the case, what was wrong with how it was before?

Reply:
I was always confused about what it was doing before but it seemed to be working so I didn't mess with it. When the thrust was increased though it became much more obvious. At first I figured it had to do with when I switched to the vector operations but those are actually equivalent to what you originally had. I actually had to go back to the vanilla code for firework rockets to make sure I wasn't missing anything. I think when you first brought that code over you made temp variables with the forward component already multiplied into the thrust and then used them instead of the thrust? When the thrust is high and you're elytra flying it is really obvious though. It'll hold you in place depending on what direction you're looking at.

Reply:
Yup, and seems like this also was related to #8015 which is convenient, as it saves me from having to look into the cause of that issue.